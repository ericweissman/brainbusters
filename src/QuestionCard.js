import React, { Component } from 'react';
import CorrectCard from './CorrectCard'
import IncorrectCard from './IncorrectCard'

class QuestionCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      answered_correctly: null,
      incorrect_answer: null
    }
  }

  evaluateGuess = (event) => {
    const {correct_answer } = this.props
    let guess = event.target.innerText;
    
    if (guess === correct_answer) {
      this.setState({
        answered_correctly: true
      })
    } else {
      this.setState({
        answered_correctly: false,
        incorrect_answer: guess
      })
    }
  }

  resetAnswered = () => {
    this.setState({
      answered_correctly: null
    })
  }

  render() {
    if (this.state.answered_correctly === null) {
      return (
        <div className='question-card' id={this.props.id}>
          <p>{this.props.question}</p>
          <div className="answer-btns">
            {
              this.props.answers.map((answer) => {
                return (
                  <button onClick={this.evaluateGuess} key={answer}>{answer}</button>
                )
              })
            }
          </div>
        </div>
      )
    } else if (this.state.answered_correctly) {
      return (
        <CorrectCard 
          question={this.props.question}
          correct_answer={this.props.correct_answer}
          id={this.props.id}
          updateGuessedCards={this.props.updateGuessedCards}
          showAllCards={this.props.showAllQuestions}
        />
      )
    } else {
      return (
        <IncorrectCard 
          question={this.props.question}
          correct_answer={this.props.correct_answer}
          incorrect_answer={this.state.incorrect_answer}
          id={this.props.id}
          updateGuessedCards={this.props.updateGuessedCards}
          updateStudyList={this.props.updateStudyList}
          showAllCards={this.props.showAllQuestions}
          reset={this.resetAnswered}
        />
      )
    }
  } 
}

export default QuestionCard;