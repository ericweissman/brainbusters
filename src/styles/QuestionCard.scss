.question-card {
  width: 500px;
  border-radius: 15px;
  border: 2px solid $dark;
  margin: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: $dark;
  color: $white;
  
  p {
    text-align: center;
    font-family: $text;
    font-weight: 800;
    padding: 10px;
  }

  button {
    @include card-button;
  }
}

.answer-btns {
  display: flex;
}

