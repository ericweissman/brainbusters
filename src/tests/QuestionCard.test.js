import React from 'react';
import QuestionCard from '../QuestionCard';
import { shallow } from 'enzyme';


const question = {
  "id": 1,
  "question": "What array prototype method would return a sum of the elements in an array?",
  "answers": [
    ".sort()",
    ".reduce()",
    ".forEach()"
  ],
  "correct_answer": ".reduce()",
  "concept": "prototypes"
}
const answers = ['a', 'b', 'c']
const correct_answer = "Testing is my favorite hobby."
const key = 1
const id = 21
const concept = "array"
const updateGuessedCardsMock = jest.fn()
const updateStudyListMock = jest.fn()
const showAllQuestions = true;

const evaluateGuessMock = jest.fn();


describe('QuestionCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <QuestionCard
        key={key}
        id={id}
        question={question}
        answers={answers}
        correct_answer={correct_answer}
        concept={concept}
        updateStudyList={updateStudyListMock}
        updateGuessedCards={updateGuessedCardsMock}
        showAllQuestions={showAllQuestions}
      />
    )
  });

  it('should match snapshot when all data is passed correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state', () => {
    expect(wrapper.state()).toEqual({answered_correctly: null, incorrect_answer: null})
  })

  // it('should fire evaluateGuess when a guess button is clicked', () => {
  //   wrapper.find('guess-button').simulate('click');
  //   expect(evaluateGuessMock).toBeCalled();
  // });

  it('should reset state of answered_correctly to null if resetAnswered is called', () => {
    wrapper.setState({ answered_correctly: true, incorrect_answer: null });
    wrapper.instance().resetAnswered();
    expect(wrapper.state()).toEqual({ answered_correctly: null, incorrect_answer: null });
  });

  // it('should change answered_correctly to True if user guess is correct when evaluate Guess is called', (event) =>{
  //   let guess = event.target.innerText
  //   guess = "array"
  //   wrapper.instance().evaluateGuess(guess);
  //   expect(wrapper.state()).toEqual({ answered_correctly: true })
  // })
})

