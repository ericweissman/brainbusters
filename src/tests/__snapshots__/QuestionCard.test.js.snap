// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionCard should match snapshot when all data is passed correctly 1`] = `
<div
  className="question-card"
  id={21}
>
  <p>
    <Component />
  </p>
  <div
    className="answer-btns"
  >
    <button
      className="guess-button"
      key="a"
      onClick={[Function]}
    >
      a
    </button>
    <button
      className="guess-button"
      key="b"
      onClick={[Function]}
    >
      b
    </button>
    <button
      className="guess-button"
      key="c"
      onClick={[Function]}
    >
      c
    </button>
  </div>
</div>
`;
